src/OpenToolkit.Windowing.Desktop/NativeWindow.cs

```diff
@@ -823,6 +823,20 @@ private void RegisterWindowCallbacks()
                 };
                 GLFW.SetJoystickCallback(_joystickCallback);
 
+                // Check for Joysticks that are connected at application launch
+                for (int i = 0, j = JoystickStates.Length; i < j; i++)
+                {
+                    if (GLFW.JoystickPresent(i))
+                    {
+                        GLFW.GetJoystickHatsRaw(i, out var hatCount);
```

Comment:
I prefer explicitly typing variables, but I decided to match the logic in previous code blocks regarding the `out var` statements

Reply:
I'd prefer if you'd do a standard loop like
`for (int i = 0; i < JoystickStates.Length; i++)` while it is a little less performant, it is better readable and the performance gain is so minimal. I mean it happens only on startup and the loop iterates 16 times. Apart from that the compiler will get better and better at one point it won't even be that little performance gain anymore.

Reply:
@jvbsl done. I'd also like to propose adding a new `public bool Present { get; }` param to JoystickState. Currently the only way to know if a joystick is actually connected is to do a comparison of `JoystickState joy != default` which I imagine isn't the best performance wise. The check is required because the joystick data is all hidden behind methods that all throw exceptions if the joystick is not actually connected:
```
public float GetAxis(int index)
        {
            return _axes[index]; <-- in default _axes is null
        }
```

Reply:
Does make total sense, go for it. Not sure on the naming though, as the event is called JoystickConnected, so perhaps IsConnected?

Reply:
the GLFW method is `GLFW.JoystickPresent` so I was thinking `Present`